docs/hugo/content/contributing/add-a-new-code-generated-resource/review-the-generated-resource.md

```diff
@@ -0,0 +1,127 @@
+---
+title: Review the generated resource
+linktitle: Review the resource
+weight: 30
+---
+
+After running the generator, source code for the new resource you added will be in the
+[v2/apis](https://github.com/Azure/azure-service-operator/blob/main/v2/api/) directory.
+
+Have a look through the files in the directory named after the `group` and `version` of the resource that was added. Source code for the custom resource will be in this folder, with supporting code in subpackages called `arm`, `storage`, and `webhook`.
+
+In our `Workspaces` example, the best place to start is `v2/api/synapse/v1api20210601/workspace_types_gen.go`
+There may be other resources that already exist in that same directory - that's expected if ASO already supported some
+resources from that provider and API version.
+
+Look for the the struct declaration representing the resource you just added.
+It should be near the top of the appropriate file, and look something like this:
+
+```go
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Ready",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="Severity",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].severity"
+// +kubebuilder:printcolumn:name="Reason",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].reason"
+// +kubebuilder:printcolumn:name="Message",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].message"
+// Generator information:
+// - Generated from: /synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/workspace.json
+// - ARM URI: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Synapse/workspaces/{workspaceName}
+type Workspace struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    Spec              Workspace_Spec   `json:"spec,omitempty"`
+    Status            Workspace_STATUS `json:"status,omitempty"`
+}
+```
+
+Look over the `Spec` and `Status` types and their properties (and the properties of their properties and so-on). For a different perspective on the same structure, review the `structure.txt` file which shows the entire structure of the resource as a tree.
+
+The Azure REST API specs from which these types are generated are not perfect, so we often have to make some adjustments. 
+
+## Check: properties that should have been marked read-only but weren't
+
+There is a section in the config detailing a number of these errors which have been fixed. Look for "Deal with properties that should have been marked readOnly but weren't".
+
+Here's an example, where DocumentDB missed setting the `provisioningState` property to read-only. We override it in the config:
+
+```yaml
+  - group: documentdb
+    name: Location  # This type is subsequently flattened into NamespacesTopics_Spec
+    property: ProvisioningState
+    remove: true
```

Comment:
I think this needs a bit more explaination i.e. why are we removing it rather than marking it as readonly or something? I assume it's because readonly properties shouldn't be included in the "Spec" as that's what's used to create the new resource. Some more context eplaining the difference between Spec and Status and how they're used by ASO would probably help. I think this ties into my high-level comment in the overview page about providing more context on what the generator actually produces etc.